optimize the unit name code (-136 bytes)

joshgoebel opened the pull request.

joshgoebel committed:
optimize writeUnitName (-130 bytes)

joshgoebel committed:
optimize readUnitName (-6 bytes)

MLXXXp commented:
The 50 ms delay is for button debounce. That's why it's done at least once.

Lets say a user presses the RIGHT button to exit the boot logo. The button bounces a few times. On the first bounce, the logo is exited. Without the delay, if the first bounce has completed when _while_ loop does its test, it will see all buttons off and continue to the sketch. If the sketch looks for a RIGHT button press early on, it may see a later bounce and do something that the user didn't intend.

Using _idle()_ in place of the delay is too short for proper debounce.

joshgoebel commented:
So is this good to go if I revert the delay?

joshgoebel commented:
I removed the commit that added idle().

MLXXXp commented:
You still need to change the _while()_ back to _do_ _while()_ for at least one button debounce delay. If you're not happy with the added delay, remove 50ms from the 750ms boot logo delay.

joshgoebel commented:
If a button was pushed in it's already been debounced IN by the delays (and fact that it had to be detected inside bootLogo)... we only need to debounce out here, which a while effectively does.

MLXXXp commented:
Not a boot logo abort with the RIGHT button.

joshgoebel committed:
Remove extra 50ms delay added by de1725

Removes the 50ms debounce delay that was added when we allow the boot
logo to be skipped by reducing the boot logo delay by 50ms.

joshgoebel commented:
Done.

MLXXXp committed:
Merge branch 'develop' into optimize_unitname
